feat: release/2.6.2

pleberre opened the pull request:
<!-- This is an auto-generated description by cubic. -->
## Summary by cubic
Release 2.6.2 adds WP Consent API integration with auto-synced consent categories shown as virtual vendors, a multilingual and customizable shortcode placeholder (incl. button text and hideable decoration), and a smoother plugin edit drawer. It also updates the signup link with UTM, bumps “Tested up to” to 6.9, fixes a PHP 8+ null pointer and a WP Consent API init bug, and improves French GTM event translations.

- **New Features**
  - WP Consent API: show 5 categories as virtual vendors and sync consent directly via wp_set_consent when the plugin is active (no GCM dependency).
  - Customizable shortcode placeholder with i18n via new meta fields and a migration (title, description, button text, hide decoration).
  - Admin UX: unified amber theme, transitions, and auto-scroll in the edit drawer.

- **Dependencies**
  - Bump webpack-dev-server to v5.2 and pin via resolutions to 5.2.1.
  - Update pbkdf2 to 3.1.3 and refresh lockfile.

<sup>Written for commit b8ccf911414445300235cbbda725147fb386ba80. Summary will update on new commits.</sup>

<!-- End of auto-generated description by cubic. -->

Loubal70 committed:
🌐 (axeptio-wordpress-plugin-fr_FR.po): improve French translation for consistency by changing "à dataLayer" to "au dataLayer" for better localization accuracy.

ogorzalka committed:
Merge pull request #19 from Loubal70/feature/69555_gtm_events

Improve french translation

Loubal70 committed:
fix: prevent null pointer error in get_plugin() for PHP 8.0+

pleberre committed:
Merge pull request #34 from axeptio/feat/74443_deprecated_variables

fix: prevent null pointer error in get_plugin() for PHP 8.0+

Loubal70 committed:
Merge branch 'develop' into feature/69555_gtm_events

Loubal70 committed:
fix: axeptio french translation

Loubal70 committed:
fix: axeptio french translation

Loubal70 committed:
Merge pull request #35 from axeptio/feature/69555_gtm_events

feat: improve gtm events translation

dependabot[bot] committed:
Bump pbkdf2 from 3.1.2 to 3.1.3

Dependabot couldn't find the original pull request head commit, df76dfd12918bdef3c15a944b212fa550c9c59f7.

Loubal70 committed:
feat: maj package versions

Loubal70 committed:
Merge pull request #13 from Loubal70/dependabot/npm_and_yarn/pbkdf2-3.1.3

Bump pbkdf2 from 3.1.2 to 3.1.3

Loubal70 committed:
fix package vulnerability

Loubal70 committed:
feat: fix laravel-mix dependencies

pleberre committed:
Merge pull request #36 from Loubal70/develop

chore: fix package.json / webpack-dev-server vulnerabilities

Loubal70 committed:
feat: ✨ Add UTM parameters to signup link

Loubal70 committed:
feat: maj readme.txt

pleberre committed:
Merge pull request #37 from axeptio/feat/75502_add_utm_sign_up

feat: ✨ Add UTM parameters to signup link

Loubal70 committed:
feat: maj readme.txt

Loubal70 committed:
feat: try change iframe to url for youtube video

pleberre committed:
Merge pull request #39 from axeptio/feat/75771_maj_readme

feat: maj readme according ticket 76771

Loubal70 committed:
feat: authorize Axeptio on WordPress version 6.9

Loubal70 committed:
feat: change XPWP_VERSION

Loubal70 committed:
feat: add WP Consent API categories integration
- Fix initialization bug (axeptioSettings → axeptioWpConsentCategories)
- Add 5 WP Consent API categories as virtual vendors
- Update consent directly without Google Consent Mode dependency
- Refactor categories data to single source of truth
- Add WordPress favicon for consent categories

Loubal70 committed:
feat: maj readme for consent api

pleberre committed:
Merge pull request #40 from axeptio/feat/77951_wordpress_6.9

feat: authorize Axeptio on WordPress version 6.9

Loubal70 committed:
feat: improve extension subpage UX (unify colors, add transitions, auto-scroll into the drawer)

Loubal70 committed:
feat: add customizable consent banner with i18n support

Loubal70 committed:
chore: remove unused require in migration

Loubal70 committed:
fix: selectedLanguages component

Loubal70 committed:
feat: replace sr-only for accessibility

pleberre committed:
Merge pull request #42 from axeptio/feat/78197_custom_context_banner

feat: PLG-31 - custom context banner

Loubal70 committed:
Merge branch 'release/2.6.2' into feat/78365_evol_consent_api

pleberre committed:
Merge pull request #41 from axeptio/feat/78365_evol_consent_api

feat: PLG-130 - add WP Consent API categories integration

claracarvalho4axeptio committed:
Merge branch 'develop' into release/2.6.2

claracarvalho4axeptio committed:
Update readme.txt

Purely cosmetic.

Co-authored-by: cubic-dev-ai[bot] <191113872+cubic-dev-ai[bot]@users.noreply.github.com>

claracarvalho4axeptio commented on package.json:
```diff
@@ -52,13 +52,16 @@
     "cross-env": "^7.0.3"
   },
+  "resolutions": {
+    "webpack-dev-server": "^5.2.1"
```
@Loubal70 , can you check this comment from cubic? Can I mark this as resolved and leave as it is?

Loubal70 replied:
This is a valid point in theory, but in our case we use BrowserSync for dev (mix watch), not Mix's built-in dev server. The webpack-dev-server version has no impact on our workflow.                                                                                                
                                                                                                                                                           
A migration from Laravel Mix to Vite.js is planned for the build pipeline, which will eliminate these dependencies. In the meantime, the build is working fine.

claracarvalho4axeptio commented:
Hey @Loubal70,

A few things to move this forward:

**1. Cubic thread to resolve**
There's still one unresolved Cubic conversation on this PR blocking the merge:
https://github.com/axeptio/wordpress-plugin/pull/38#discussion_r2873739515

Could you take a look and resolve it when you get a chance? Once that's cleared, we should be good to merge `release/2.6.2` into `develop`.

**2. Next steps after merge — I need to understand the workflow**
Since this is my first release with you in this repo, I want to make sure I follow the right process:

a. After merging into `develop`, will we need to create a separate PR to merge `release/2.6.2` into `main` (production)? Or do we merge `develop` into `main` at that point?

b. What is the environment at https://wordpress-843309-3452223.cloudwaysapps.com/wp-admin — is that staging/preprod? And who is responsible for deploying there? I want to push this release into production.

Thanks!

Loubal70 commented:
Hi,

Thanks for the detailed message. I'm currently on vacation, so I won't be able to look into this right away — I'll try to take a quick look tomorrow morning though.

Regarding the PR, I've set it to draft for now. We're waiting on @ogorzalka , who needs to set up automatic deployment and document the release process. At the moment, he's the only person able to push a release to production.

As for the workflow, I'd leave it up to Olivier to define the most appropriate approach. Personally, I think we could use `main` as the public/production branch, and either drop `develop` entirely or keep it as a base for new feature branches.

About the environment at the Cloudways URL you mentioned — it's currently managed by Amphibee, and it doesn't have an automated deployment pipeline either. The plan is to set up automatic deployment when a PR is merged into the preprod branch (which will need to be protected, of course).

I'll get back to you soon.

Best,
Louis

Loubal70 committed:
feat: hide consent tag in extensions page when "consentRequirements" is empty in axeptio api

claracarvalho4axeptio committed:
Merge pull request #49 from axeptio/feat/74444_cf7

fix: filter vendors without consent requirements from recommended settings

ogorzalka committed:
chore: merge develop into release/2.6.2

ogorzalka committed:
fix: resolve webpackbar compatibility with webpack 5.106+

Force webpackbar ^7.0.0 via resolutions to fix ProgressPlugin
invalid options error (name, color, reporters) introduced by
webpackbar 5.x/6.x being incompatible with recent webpack 5.

ogorzalka committed:
fix: resolve JS errors and improve frontend robustness

- Use Axeptio_SDK instead of axeptioSettings to avoid undefined
  reference due to script execution order
- Use window.wp_set_consent to fix no-undef ESLint error
- Return empty string for non-default languages in getLocalizedMeta
  fallback to allow PO/MO translation fallback

ogorzalka committed:
fix: address PR review feedback

- Fix missing spaces in readme.txt formatting
- Add rel="noopener noreferrer" to target="_blank" links in onboarding
- Remove fuzzy flags on support URL translations (de, en, es, fr)

ogorzalka committed:
fix: resolve PHPCS violations

- Replace DROP COLUMN IF EXISTS with INFORMATION_SCHEMA checks for
  MySQL 5.7 compatibility in migration 2.6.2
- Prefix template variables with axeptio_ to satisfy PrefixAllGlobals
- Add phpcs:disable/enable blocks for migration schema queries
- Auto-fix formatting and coding standards violations

ogorzalka committed:
Merge branch 'develop' into release/2.6.2

ogorzalka committed:
fix: preserve legacy plain-string meta values when editing localized fields

When setLocalizedMeta converts a plain string to JSON for multilingual
storage, preserve the existing value under the 'default' key to prevent
data loss for pre-existing installations.
